*: move server/status -> server/status/statuspb

jordanlewis opened the pull request:
This allows dependents of the protobufs in server/status to not have to
depend on server/status.

Also, stop depending on `status` from `sql` - it just needed one method on `status.MetricsRecorder`, which we replace with a package-local narrowed interface.

Update #30001.

Release note: None

jordanlewis committed:
*: move server/status -> server/status/statuspb

This allows dependents of the protobufs in server/status to not have to
depend on server/status.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31309)
<!-- Reviewable:end -->

jordanlewis committed:
sql: remove dependency on server/status

We were importing server/status to get the MetricsRecorder struct, but
we only use one method on it. Remove that edge by adding a narrowed
interface, local to sql.

Release note: None

vilterp commented on pkg/server/serverpb/admin.proto:
```diff
@@ -22,6 +22,7 @@ import "server/serverpb/status.proto";
 import "storage/engine/enginepb/mvcc.proto";
 import "storage/liveness.proto";
 import "storage/log.proto";
+
```
nit: remove blank line

jordanlewis commented:
(got merged in another PR)
